ContinousBenchmarking.md

```diff
@@ -0,0 +1,88 @@
+Continuous Benchmarking System
+======
+Kevin Cheng
+Initial Proposal 11/6/18
+
+
+# Background:
+## Context
+Benchmarking an algorithm is a compute intensive and time consuming chore that is essential for guaranteeing performance. 
+An algorithm can unknowingly drift towards an undesirable state if the laborious task of benchmarking is neglected for
+too long.  
+
+## Problem Statement
+We want to create a continous benchmarking system to automate this task.  
+
+## Scope
+Problem starts with a code changes to Garage. The system needs to run benchmarking at least nightly, generate a results page, and deploy that page.  Problem ends with a pubically viewable results page.
+
+
+## Goals and Non-Goals
+*This version of the proposal will focus on a minimally viable system. These goals will evolve once the initial goal is accomplished and we decide to bring in more complexity.
+#### Goals:
+- Update benchmarking scripts so they can be started in a consistent format and run to completion without manual intervention.
+- Focus on getting system up and running on a single machine.
+- Design a simple system for deploying the results page.
+- Allow garage developers to detect regressions in algorithm performance caused by code changes
```

Comment:
how specific do you want to be with 'code changes'? Just specify which commit is causing regressions or something more specific?

Reply:
Yes I think that is granular enough.